### PR TITLE
removing deprecated scheduler metrics

### LINKED ISSUE #94852: Remove deprecated scheduler metrics
Remove all metrics marked as deprecated in https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/metrics/metrics.go#L135

/sig scheduling
/cc @logicalhan 
/help

### PR DESCRIPTION
**Pull request by arghya88**
**What type of PR is this?**

/kind feature
/sig scheduling

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #94852

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Removes deprecated scheduler metrics DeprecatedSchedulingDuration, DeprecatedSchedulingAlgorithmPredicateEvaluationSecondsDuration, DeprecatedSchedulingAlgorithmPriorityEvaluationSecondsDuration
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/ok-to-test
wait for clearing ci

**Review by draveness — COMMENTED** [review]
/priority backlog

**Review by chendave — COMMENTED** [review]
I think this is an end-user facing change, release note is preferred for this change.

**Comment by arghya88** [comment]
@chendave Added release note

File: pkg/scheduler/core/generic_scheduler.go [hunk 1]
```diff
@@ -157,28 +156,11 @@ func (g *genericScheduler) Schedule(ctx context.Context, prof *profile.Profile,
...
 	}
 
-	metrics.DeprecatedSchedulingAlgorithmPredicateEvaluationSecondsDuration.Observe(metrics.SinceInSeconds(startPredicateEvalTime))
-	metrics.DeprecatedSchedulingDuration.WithLabelValues(metrics.PredicateEvaluation).Observe(metrics.SinceInSeconds(startPredicateEvalTime))
-
-	startPriorityEvalTime := time.Now()
-	// When only one node after predicate, just use it.
-	if len(feasibleNodes) == 1 {
-		metrics.DeprecatedSchedulingAlgorithmPriorityEvaluationSecondsDuration.Observe(metrics.SinceInSeconds(startPriorityEvalTime))
-		return ScheduleResult{
```
Comment (adtac) [inline pkg/scheduler/core/generic_scheduler.go:167]: this shouldn't be removed -- the scheduler should still return immediately when there's only one feasible node

**Comment by adtac** [comment]
cc @ahg-g for milestone

File: pkg/scheduler/core/generic_scheduler.go (hunk 1 above)
Comment (arghya88) [reply pkg/scheduler/core/generic_scheduler.go:167]: Thanks for catching that..fixed it

File: cmd/kube-scheduler/app/server.go [hunk 2]
```diff
@@ -233,7 +232,6 @@ func installMetricHandler(pathRecorderMux *mux.PathRecorderMux) {
 	defaultMetricsHandler := legacyregistry.Handler().ServeHTTP
 	pathRecorderMux.HandleFunc("/metrics", func(w http.ResponseWriter, req *http.Request) {
 		if req.Method == "DELETE" {
-			metrics.Reset()
```
Comment (ahg-g) [inline cmd/kube-scheduler/app/server.go:236]: why is this removed?

File: pkg/scheduler/metrics/metrics.go [hunk 3]
```diff
@@ -28,8 +28,6 @@ import (
 const (
 	// SchedulerSubsystem - subsystem name used by scheduler
 	SchedulerSubsystem = "scheduler"
-	// DeprecatedSchedulingDurationName - scheduler duration metric name which is deprecated
-	DeprecatedSchedulingDurationName = "scheduling_duration_seconds"
```
Comment (ahg-g) [inline pkg/scheduler/metrics/metrics.go:32]: remove `PredicateEvaluation`, `PriorityEvaluation` below as well

**Review by ahg-g — COMMENTED** [review]
thanks

File: cmd/kube-scheduler/app/server.go (hunk 2 above)
Comment (arghya88) [reply cmd/kube-scheduler/app/server.go:236]: I removed DeprecatedSchedulingDuration from metrics.go and because of that removed Reset() as well because it was calling DeprecatedSchedulingDuration.Reset()

Comment (ahg-g) [reply cmd/kube-scheduler/app/server.go:236]: ah, sounds good then

File: pkg/scheduler/metrics/metrics.go (hunk 3 above)
Comment (arghya88) [reply pkg/scheduler/metrics/metrics.go:32]: Done
